### PR TITLE
[data] Add iterator batch_format=None support, which will yield batches in the current batch format with zero copies

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This PR is a cleanup of https://github.com/ray-project/ray/pull/33536

It uses "None" instead of "zero-copy" as a batch format, since None has a similar meaning for batch_size, where it means a system-chosen batch size. Here "None" also means the system chosen optimal batch format.

### CODE REVIEW

File: python/ray/data/dataset.py [hunk 1]
```diff
@@ -379,7 +379,7 @@ def map_batches(
         *,
         batch_size: Optional[Union[int, Literal["default"]]] = "default",
         compute: Optional[Union[str, ComputeStrategy]] = None,
-        batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
+        batch_format: Optional[str] = "default",
```
Comment (amogkam) [inline python/ray/data/dataset.py:382]: keep it as `Optional[Literal]` for full explicitness of supported batch formats?

**Review by amogkam — APPROVED** [review]

**Comment by amogkam** [comment]
lets also keep the documentation changes from https://github.com/ray-project/ray/pull/33536/files#diff-988f3832ac94d085daf61260175e2580920ebd1521dc760f58b426b94379d5b7L235?

File: python/ray/data/dataset.py (hunk 1 above)
Comment (ericl) [reply python/ray/data/dataset.py:382]: I feel like that is hard to maintain (per the inconsistencies in the code already), so opted to go unify on the shorter signature.

**Comment by ericl** [comment]
> lets also keep the documentation changes from https://github.com/ray-project/ray/pull/33536/files#diff-988f3832ac94d085daf61260175e2580920ebd1521dc760f58b426b94379d5b7L235?


Done

**Review by clarkzinzow — APPROVED** [review]

**Pull request merged**